react/features/dynamic-branding/middleware.native.js

```diff
@@ -0,0 +1,49 @@
+import { SET_CONFIG } from '../base/config';
+import { CONNECTION_DISCONNECTED, CONNECTION_FAILED } from '../base/connection';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { SET_DYNAMIC_BRANDING_DATA } from './actionTypes';
+import { fetchCustomBrandingData, unsetDynamicBranding } from './actions.native';
+
+
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case CONNECTION_FAILED: {
+        store.dispatch(unsetDynamicBranding());
+
+        return next(action);
+    }
+
+    case SET_CONFIG: {
+        const result = next(action);
+
+        store.dispatch(fetchCustomBrandingData());
+
+        return result;
+    }
+
+    case SET_DYNAMIC_BRANDING_DATA: {
+        const {
+            avatarBackgrounds,
+            backgroundColor,
+            backgroundImageUrl
+        } = action.value;
+
+        action.value.avatarBackgrounds = avatarBackgrounds.filter(
+            color => !color.includes('linear-gradient')
```

Comment:
Let's try not to ship this and have the customer remove it from their branding instead, while we work on adding support for it.